src/runtime_src/driver/xclng/drm/xocl/subdev/mailbox.c

```diff
@@ -1426,6 +1449,9 @@ int mailbox_request(struct platform_device *pdev, void *req, size_t reqlen,
 
 	/* Kick TX channel to try to send out msg. */
 	complete(&mbx->mbx_tx.mbc_worker);
+	wait_for_completion(&reqmsg->mbm_complete);
+	rx_enqueued_msg_timer_on(mbx, reqmsg->mbm_req_id);
```

Comment:
do you mean respmsg? and only turn on timer when reqmsg was successfully sent out?